dynamic/clientset/dynamic_clientset.go

```diff
@@ -64,6 +64,15 @@ func (cs *Clientset) Kind(apiVersion, kind, namespace string) (*ResourceClient,
 	return cs.resource(apiResource, namespace)
 }
 
+func getResourceGVR(apiResource *dynamicdiscovery.APIResource) schema.GroupVersionResource {
+	resource := apiResource.APIResource
+	return schema.GroupVersionResource{
+		Group:    resource.Group,
+		Version:  resource.Version,
+		Resource: resource.Name,
+	}
+}
+
 func (cs *Clientset) resource(apiResource *dynamicdiscovery.APIResource, namespace string) (*ResourceClient, error) {
```

Comment:
One of the improvements in the new dynamic client is that it decouples specification of the namespace from the creation of the client. We should mirror that by removing the `namespace` argument here and in `Clientset.Resource()`.

Reply:
Oh yeah, that one too. Good catch. :)